File: Connectors/InteractiveBrokers/Java/Client/UnderComp.java
/* Copyright (C) 2013 Interactive Brokers LLC. All rights reserved.  This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

package com.ib.client;

public class UnderComp {

	public int    m_conId;
	public double m_delta;
	public double m_price;

	public UnderComp() {
		m_conId = 0;
		m_delta = 0;
		m_price = 0;
	}

    public boolean equals(Object p_other) {

    	if (this == p_other) {
    		return true;
    	}

    	if (p_other == null || !(p_other instanceof UnderComp)) {
    		return false;
    	}

        UnderComp l_theOther = (UnderComp)p_other;

        if (m_conId != l_theOther.m_conId) {
        	return false;
        }
        if (m_delta != l_theOther.m_delta) {
        	return false;
        }
        if (m_price != l_theOther.m_price) {
        	return false;
        }

        return true;
    }
}

File: Connectors/InteractiveBrokers/CSharp/UnderComp.cs
/* Copyright (C) 2013 Interactive Brokers LLC. All rights reserved.  This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;

namespace IBApi
{
    /**
     * @brief Delta-Neutral Underlying Component.
     */
    public class UnderComp
    {
        private int conId;
        private double delta;
        private double price;

        /**
         * @brief
         */
        public int ConId
        {
            get { return conId; }
            set { conId = value; }
        }

        /**
        * @brief
        */
        public double Delta
        {
            get { return delta; }
            set { delta = value; }
        }

        /**
        * @brief
        */
        public double Price
        {
            get { return price; }
            set { price = value; }
        }
    }
}
